src/dummy.rs

```diff
@@ -176,6 +180,108 @@ impl Runnable for DummyVM {
         }
         Ok(res)
     }
+
+    fn run(cfg: ErgConfig) {
+        let quiet_repl = cfg.quiet_repl;
+        let mut instance = Self::new(cfg.clone());
+        let res = match instance.input() {
+            Input::File(_) | Input::Pipe(_) | Input::Str(_) => instance.exec(),
+            Input::REPL => {
+                let output = stdout();
+                let mut output = BufWriter::new(output.lock());
+                if !quiet_repl {
+                    log!(info_f output, "The REPL has started.\n");
+                    output
+                        .write_all(instance.start_message().as_bytes())
+                        .unwrap();
+                }
+                output.write_all(instance.ps1().as_bytes()).unwrap();
+                output.flush().unwrap();
+                let mut in_block = false;
+                let mut parser = ParserRunner::new(cfg);
```

Comment:
parser is declared here, and I am using it, but it is not possible to add this to the Runnable trait.
Or is there a way around this?

Reply:
`expect_nextline` should be declared with no body, and the specific judgment algorithm should be defined in each struct. Compiler, Parser etc. uses Parser, Lexer should simply return `false`.

Reply:
Thank you!!
I'll try it